src/vs/platform/actions/common/actions.ts

```diff
@@ -388,7 +390,9 @@ export class MenuItemAction implements IAction {
 		@ICommandService private _commandService: ICommandService
 	) {
 		this.id = item.id;
-		this.label = typeof item.title === 'string' ? item.title : item.title.value;
+		this.label = options?.renderShortTitle && item.shortTitle
+			? (typeof item.shortTitle === 'string' ? item.shortTitle : item.shortTitle.value)
+			: (typeof item.title === 'string' ? item.title : item.title.value);
```

Comment:
Not brilliant but inherent because  `ICommandAction` must fit "onto" `IAction`. That by itself is debt because there is no real reason for using `IAction`, it's kinda glue for rendering (toolbar, action bar etc)